src/kili/core/utils/pagination.py

```diff
@@ -115,3 +116,14 @@ def generate_variables(batched_properties):
     if batch and results and last_batch_callback:
         last_batch_callback(batch, results)
     return results
+
+
+def batcher(iterable: Iterator, batch_size: int):
+    """Break iterable into sub-iterables with batch_size elements each.
+
+    The last yielded list will have fewer than n elements if the
+    length of iterable is not divisible by batch_size:
+    """
+    iterator = iter(iterable)
+    while batch := list(islice(iterator, batch_size)):
+        yield batch
```

Comment:
we already have a BatchIteratorBuilder in core/utils/pagination

Reply:
It only work for list because of the __len__ function.
This one is a simpler one and works for generators